flyteidl/clients/go/admin/token_source_provider.go

```diff
@@ -120,15 +120,24 @@ func NewExternalTokenSourceProvider(command []string) (TokenSourceProvider, erro
 }
 
 func (e ExternalTokenSourceProvider) GetTokenSource(ctx context.Context) (oauth2.TokenSource, error) {
-	output, err := externalprocess.Execute(e.command)
+	return &externalCommandTokenSource{
+		command: e.command,
+	}, nil
+}
+
+type externalCommandTokenSource struct {
+	command []string
+}
```

Comment:
Do we really need to define `externalCommandTokenSource`? We could implement `Token` in `ExternalTokenSourceProvider`, right?

Reply:
We could implement `Token()` in `ExternalTokenSourceProvider`, but I feel it would potentially feel a bit confusing to follow.
I don't personally mind either way.